main.cpp

```diff
@@ -128,7 +142,19 @@ struct Numeric
 
     template<typename ArgType>
     explicit Numeric (ArgType initValue) : value(std::make_unique<Type>(static_cast<NumericType>(initValue))) {}
```

Comment:
See if you can figure out how to make this constructor not be templated.  
The constructor argument should simply be `NumericType`, not `ArgType`
- [ ] revise